### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference - go.mongodb.org/mongo-driver/x/mongo/driverlegacy.KillCursors

### ISSUE THREAD

**Issue by yudao** [rwynn/monstache#245]
Hi!

I have this message all the time now in monstache error log.

I don't know if it's a monstache or mongodb problem.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0xf01e5c]

goroutine 71 [running]:
go.mongodb.org/mongo-driver/x/mongo/driverlegacy.KillCursors(0x1432900, 0xc000552a20, 0xc0003d0ea0, 0x5, 0xc0003d0ea6, 0xa, 0xc0000a8500, 0x7566ad16c16003f7, 0x0, 0x0, ...)
	/home/vagrant/go/pkg/mod/go.mongodb.org/mongo-driver@v1.0.1-0.20190507231345-c7d9b5376a19/x/mongo/driverlegacy/kill_cursors.go:35 +0x2cc
go.mongodb.org/mongo-driver/mongo.(*ChangeStream).loopNext(0xc00011ca00, 0x1432900, 0xc000552a20)
	/home/vagrant/go/pkg/mod/go.mongodb.org/mongo-driver@v1.0.1-0.20190507231345-c7d9b5376a19/mongo/change_stream.go:556 +0x354
go.mongodb.org/mongo-driver/mongo.(*ChangeStream).Next(0xc00011ca00, 0x1432900, 0xc000552a20, 0x1432900)
	/home/vagrant/go/pkg/mod/go.mongodb.org/mongo-driver@v1.0.1-0.20190507231345-c7d9b5376a19/mongo/change_stream.go:505 +0x169
github.com/rwynn/gtm.ConsumeChangeStream(0xc0004c3320, 0xc0004a22c0, 0xc0000b0f00, 0x10, 0xc0001ae620, 0x0, 0x0)
	/home/vagrant/go/pkg/mod/github.com/rwynn/gtm@v0.0.0-20190510014426-6a4f37ffe043/gtm.go:1205 +0x211
created by github.com/rwynn/gtm.Start
	/home/vagrant/go/pkg/mod/github.com/rwynn/gtm@v0.0.0-20190510014426-6a4f37ffe043/gtm.go:1708 +0x396
```

Thx!

Yu.

**Comment by rwynn** [comment]
It looks like you have an old mongodb driver @v1.0.1.  The latest monstache uses the driver @v1.0.3 (the latest).  Can you try the latest version of monstache?

**Comment by rwynn** [comment]
Sorry meant to say use the latest version of monstache that is compatible with your version of Elasticsearch.  That would be monstache v5.0.8 for Elasticsearch 6 and monstache v6.0.8 for Elasticsearch 7.

**Comment by rwynn** [comment]
By the way I think this particular panic was fixed in the golang driver @v1.0.2.

https://jira.mongodb.org/browse/GODRIVER-989

**Comment by yudao** [comment]
@thx a lot!
